appcelerator/titanium_mobile
# fix(android): decimal comma TextField handling on some devices

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-28312

**Summary:**
- Some devices such as Samsung and Huawei do not have localized decimal keyboards and always show a `.` for the decimal separator. This prevents users in locales such as Europe from entering a decimal separator because the `.` is filtered-out.
- This is never an issue in the Android emulator or Google's devices.
- This PR will substitute a `.` with `,`. Unfortunately there doesn't appear too be any other solution.

**Note:**
This should be tested by @lokeshchdhry since he has a device that can reproduce this issue.

**Test:**
1. Acquire a Samsung or Huawei Android device.
2. Go to: `Settings -> System -> Languages & Input -> Languages`
3. Select a European language such as French or German.
4. Build and run the below code.
5. Verify TextField displays `1.234,5`. (Will be `1 234,5` for French.)
5. Tap on the TextField.
6. Delete all characters.
7. Attempt to enter `1,5` or `1.5`.
8. Verify it displays as `1,5`. (Tapping a period `.` will show a comma `,` instead.)

```javascript
const value = 1234.5;
const window = Ti.UI.createWindow();
const textField = Ti.UI.createTextField({
	value: value.toLocaleString(Ti.Locale.currentLocale),
	keyboardType: Ti.UI.KEYBOARD_TYPE_DECIMAL_PAD,
	width: "80%",
});
textField.addEventListener("change", (e) => {
	console.log("@@@ value: " + Ti.Locale.parseDecimal(e.value));
});
window.add(textField);
window.open();
```
